pandas/tests/plotting/test_datetimelike.py

```diff
@@ -1271,6 +1272,22 @@ def test_plot_outofbounds_datetime(self):
         values = [datetime(1677, 1, 1, 12), datetime(1677, 1, 2, 12)]
         self.plt.plot(values)
 
+    def test_format_timedelta_ticks(self):
+        import matplotlib.pyplot as plt
+        rng = timedelta_range('0', periods=3, freq='ns')
+        df = DataFrame(np.random.randn(len(rng), 3), rng)
+        ax = df.plot()
+        plt.gcf().autofmt_xdate()
+        xaxis = ax.get_xaxis()
+        for l in xaxis.get_ticklabels():
+            if len(l.get_text()) > 0:
+                self.assertEqual(l.get_rotation(), 30)
+
```

Comment:
@TomAugspurger do you think you can help me finish this test. I'm not sure how best to do it and I have to get back to what I wrote this for today. We basically want to check that for a narrow window:

```python
rng = timedelta_range('0', periods=3, freq='ns')
```

That we get nanosecond precision, and if at 0 to start without days.

We also want to check for a wide window that we have second precision.

```
rng = timedelta_range('0', periods=10, freq='1 d')
```